### PR TITLE
[czmq] Update to v4.2.1

### PR DESCRIPTION
**Pull request by myd7349**
**Describe the pull request**

- #### What does your PR fix?  
  Fixes #...

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  <all / linux, windows, ...>, <Yes/No>

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  <Yes / I am still working on this PR>

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Comment by JonLiu1993** [comment]
@myd7349  Ready for review now?

**Comment by JonLiu1993** [comment]
@myd7349, thanks for your contribution! ping us if the PR ready for review.

**Comment by JonLiu1993** [comment]
@myd7349, thanks for your contribution! ping us if the PR ready for review.

**Comment by myd7349** [comment]
Hi! @JonLiu1993 @PhoebeHui Sorry for the long delay.
This PR is ready now. Please review, thanks.

File: ports/czmq/portfile.cmake [hunk 1]
```diff
@@ -44,20 +44,20 @@ vcpkg_configure_cmake(
...
 vcpkg_copy_pdbs()
 
 if(EXISTS ${CURRENT_PACKAGES_DIR}/CMake)
-    vcpkg_fixup_cmake_targets(CONFIG_PATH CMake)
+    vcpkg_cmake_config_fixup(CONFIG_PATH CMake)
 endif()
 if(EXISTS ${CURRENT_PACKAGES_DIR}/share/cmake/${PORT})
-    vcpkg_fixup_cmake_targets(CONFIG_PATH share/cmake/${PORT})
+    vcpkg_cmake_config_fixup(CONFIG_PATH share/cmake/${PORT})
 endif()
```
Comment (JonLiu1993) [inline ports/czmq/portfile.cmake:57]:
```suggestion
vcpkg_fixup_pkgconfig()
```

File: ports/czmq/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,53 @@
+{
+  "name": "czmq",
+  "version": "4.2.1",
```
Comment (JonLiu1993) [inline ports/czmq/vcpkg.json:3]:
```suggestion
  "version-semver": "4.2.1",
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by JonLiu1993** [comment]
@myd7349 ，
Thanks for your pr.
have you tested the feature locally on x64-linux, x64-windows, and x64-windows-static？

**Comment by myd7349** [comment]
> 
> 
> @myd7349 ，
> Thanks for your pr.
> have you tested the feature locally on x64-linux, x64-windows, and x64-windows-static？

Hi! @JonLiu1993 Yes. I have tested the features locally.

**Comment by JonLiu1993** [comment]
> > @myd7349 ，
> > Thanks for your pr.
> > have you tested the feature locally on x64-linux, x64-windows, and x64-windows-static？
> 
> Hi! @JonLiu1993 Yes. I have tested the features locally.

Please complete the modification I requested and I can approve this pr

**Comment by myd7349** [comment]
> 
> 
> > > @myd7349 ，
> > > Thanks for your pr.
> > > have you tested the feature locally on x64-linux, x64-windows, and x64-windows-static？
> > 
> > 
> > Hi! @JonLiu1993 Yes. I have tested the features locally.
> 
> Please complete the modification I requested and I can approve this pr

Hi! @JonLiu1993 Thanks!

**Review by JonLiu1993 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for your contribution!

**Pull request merged**